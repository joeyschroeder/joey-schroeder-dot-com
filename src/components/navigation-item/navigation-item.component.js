import React, { Component } from 'react';

import { CHILDREN_PROP_TYPE } from '../../constants/children-prop-type';
import { Icon } from '../icon/icon.component';
import PropTypes from 'prop-types';
import styles from './navigation-item.style.scss';

export class NavigationItem extends Component {
  static propTypes = {
    active: PropTypes.bool,
    children: CHILDREN_PROP_TYPE,
    href: PropTypes.string,
    iconName: PropTypes.string
  };

  static defaultProps = {
    active: false,
    children: null,
    href: '',
    iconName: ''
  };

  getStyle() {
    const { active } = this.props;

    const translateX = active ? 0 : '100%';
    const transform = `translateX(${translateX})`;

    const style = { transform };

    return style;
  }

  render() {
    const { href, children, iconName } = this.props;
    const style = this.getStyle();

    return (
      <a className={styles.root} style={style} href={href} target="_blank" rel="noopener noreferrer">
        <div style={{ marginRight: '1rem' }}>
          <Icon className={iconName} />
        </div>
        <div style={{ flex: 1 }}>{children}</div>
      </a>
    );
  }
}
