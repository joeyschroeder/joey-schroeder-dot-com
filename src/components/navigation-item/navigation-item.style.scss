@import "../../styles/config/variables";
@import "../../styles/mixins/transition";
@import "../../styles/mixins/font-size";
@import "../../styles/mixins/line-height";

.root {
  @include font-size(1.8);
  @include line-height(1.8);
  @include transition(transform);

  border-bottom-style: solid;
  border-bottom-width: $border-width;
  border-bottom-color: $color-stripe-1;
  color: $color-white;
  display: flex;
  font-weight: $font-weight-light;
  padding: $spacer * 2;
  text-decoration: none;
  text-transform: uppercase;

  @for $i from 1 through 10 {
    &:nth-child(#{$i}) {
      transition-delay: $i * $duration-base / 3;
    }
  }

  &:nth-child(1) {
    border-color: $color-stripe-4;
  }

  &:nth-child(2) {
    border-color: $color-stripe-3;
  }

  &:nth-child(3) {
    border-color: $color-stripe-2;
  }

  &:nth-child(4) {
    border-color: $color-stripe-1;
  }

  @media screen and (max-width: $width-mobile) {
    padding-top: $spacer * 1.5;
    padding-bottom: $spacer * 1.5;
  }
}
