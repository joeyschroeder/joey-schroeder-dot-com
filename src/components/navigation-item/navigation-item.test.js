import React from 'react';
import { create } from 'react-test-renderer';
import { NavigationItem } from './navigation-item.component';

describe('<NavigationItem />', () => {
  it('should render correctly', () => {
    const component = create(<NavigationItem />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
