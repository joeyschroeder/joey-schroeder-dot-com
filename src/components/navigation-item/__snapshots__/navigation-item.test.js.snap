// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItem /> should render correctly 1`] = `
<a
  className="root"
  href=""
  rel="noopener noreferrer"
  style={
    Object {
      "transform": "translateX(100%)",
    }
  }
  target="_blank"
  title=""
>
  <div
    style={
      Object {
        "marginRight": "1rem",
      }
    }
  >
    <i
      aria-hidden="true"
      className=""
    />
  </div>
  <div
    style={
      Object {
        "flex": 1,
      }
    }
  >
    
  </div>
</a>
`;
